tests/run-with-compiler/i14541.scala

```diff
@@ -0,0 +1,8 @@
+
+object Test:
```

Comment:
Found a way to properly test `MainGenericRunner.main`
```scala
@main def Test: Unit =
  val classpath = dotty.tools.dotc.util.ClasspathFromClassloader(getClass.getClassLoader)
  dotty.tools.MainGenericRunner.main(Array("-classpath", classpath, "echo", "hello", "world"))
```

Reply:
@som-snytt could you double-check that works as expected with your change and fails without it?

Reply:
@som-snytt did that work?

Reply:
I'll try it out this morning.

Reply:
Nice.